modules/cloud-run-v2/variables.tf

```
@@ -125,6 +125,41 @@ variable "iam" {
   default     = {}
 }
 
+variable "iap_config" {
+  description = "If present, turns on Identity-Aware Proxy (IAP) for the Cloud Run service."
+  type = object({
+    iam          = optional(list(string))
+    iam_additive = optional(list(string))
+  })
+  default = null
+
+  validation {
+    condition = var.iap_config == null || (
+      (try(var.iap_config.iam, null) != null && try(var.iap_config.iam_additive, null) == null) ||
+      (try(var.iap_config.iam, null) == null && try(var.iap_config.iam_additive, null) != null)
+    )
+    error_message = "When iap_config is provided, exactly one of 'iam' or 'iam_additive' must be specified."
+  }
+
+  validation {
+    condition = var.iap_config == null || (
+      (try(var.iap_config.iam, null) != null ? length(var.iap_config.iam) > 0 : true) &&
+      (try(var.iap_config.iam_additive, null) != null ? length(var.iap_config.iam_additive) > 0 : true)
+    )
```

```suggestion
    condition = !(length(try(var.iap_config.iam, [])) > 0 && length(try(var.iap_config.iam_additive, [])) > 0)
    )
```
This might be easier to cover by negative condition - we just don't want to have both lists non-empty